Demo2 Fix tests

AtrisFlax opened the pull request.

AtrisFlax committed:
fix some tests

AtrisFlax committed:
Filters + Event tests

AtrisFlax committed:
delete not using init

Romarub commented on ItHappened.Application/Services/EventService/EventService.cs:
```diff
@@ -54,7 +55,8 @@ public Event GetEvent(Guid actorId, Guid eventId)
         {
-            return EventsFilter.Filter(GetAllEvents(actorId, trackerId), eventsFilters);
+            var filteredEvents = EventsFilter.Filter(GetAllEvents(actorId, trackerId), eventsFilters);
+            return filteredEvents.ToList().AsReadOnly();
```
AsReadOnly мне кажется тут излишне. Метод сам возвращает ReadOnly коллекцию. 
Там это автоматически преобразуется судя по возвращаемым типам из подобных методов.

Romarub commented on ItHappened.Domain/EventTracker/EventTracker.cs:
```diff
@@ -17,19 +17,23 @@ public EventTracker(Guid id, Guid creatorId, string name, TrackerCustomizationSe
 
 
-        private bool CustomizationMatch(bool isRequired, bool isEventSome)
+        private bool CustomizationMatch(bool isTrackerRequired, bool isEventHas)
```
Степе писал про это. Предлагаю придерживаться правилам нейминга из Чистого кода.
Если функция возвращает значение bool, то там такие функции начинаются с Is. т.е. IsCustomizationMatch и тп.
Так сразу из названия понятно, что функция возвращает, без надобности обращаться к самой функции.
см. Чистый код, глава 2, Имена методов

Romarub commented on ItHappened.Domain/Statistics/Calculators/BestRatingEventCalculator.cs:
```diff
@@ -8,38 +8,43 @@ namespace ItHappened.Domain.Statistics
+            if (bestRatingEventInfo.Event.HappensDate > now.AddDays(-MinDaysThreshold))
+            {
+                 return Option<ISingleTrackerFact>.None;
+            }
```
Третье условие функции применимости проверяется не в методе CanCalculate. Это неправильно.

Romarub commented on ItHappened.Domain/Statistics/Calculators/BestRatingEventCalculator.cs:
```diff
@@ -48,21 +53,40 @@ var bestRatingEventInfo
-                .Count(@event => @event.CustomizationsParameters.Rating.IsSome) >= TresholdEventsWithRating;
-            MaxDateThreshold = 90;
-            var isOldestEventHappenedMoreThanThreeMonthsAgo = events
+            var isRatherEventHappenedMoreThanThreeMonthsAgo = events
```
Тут с неймингом проблемы. В переменной содержится дата, а называется как bool. И сам смысл незвания непонятен. Rather тут неприменимо. Earliest или Oldest. Я поправлю потом, как функция CanCalculate будет нормальная

Romarub commented on ItHappened.Domain/Statistics/Calculators/WorstRatingEventCalculator.cs:
```diff
@@ -1,58 +1,59 @@
             {
                  return Option<ISingleTrackerFact>.None;
             }
 
```
Аналогично замечаниям с BestRatingCalculator

Romarub reviewed (changes requested):
смотри комменты

AtrisFlax committed:
Fix for review

AtrisFlax committed:
Fix worst\best event.

CanCalculate has all applicability function checks.
Change applicability function, one of events should be 3 months ago

AtrisFlax committed:
#issue 129

Added uniform DateTimeOffset.Not to multiple calculators
